### ISSUE TITLE
Accessibility: Skip to content link does not work

### ISSUE THREAD

**Issue by spacedmonkey** [swissspidy/web-stories-wp#11055]
## Bug Description

When using keyboard navigation, a "Skip to content" link is shown. When pressing enter, the url changes and but nothing happens. Worse in dashboard, the content is cleared. See video. 

## Screenshots

https://user-images.githubusercontent.com/237508/159733308-af0bfb75-a879-424a-b856-2bf48da1ca12.mov

https://user-images.githubusercontent.com/237508/159733286-a288b33d-2b43-4c80-966d-7cfd05c2a4d8.mov

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

**Comment by swissspidy** [comment]
From an accessibility perspective the skip link (which comes from WP, FWIW) seems to work as expected. It skips the WordPress admin menu so that the next time you press <kbd>Tab</kbd> you're within the dashboard/editor already.

What is broken is the empty content on the dashboard, which I assume is because the dashboard takes the `#wpbody-content` hash and tries (and fails) to find a matching route.

You get the same experience when opening something like `/wp-admin/edit.php?post_type=web-story&page=stories-dashboard#/foobar` directly.

To fix this, I think we could just redirect to the dashboard if there is no matching route.

cc @BrittanyIRL

**Comment by BrittanyIRL** [comment]
@felipebochehin87 

Testing directions copied from PR: 
> Head to the Stories Dashboard and verify that Dashboard loads by default, that "Explore Templates" and "Template Details" still work as expected.
Now use the keyboard to navigate to "Main Content" skip button (tab through WP admin toolbar, if you just loaded the page it'll be the first tab). Hit "Enter" on "Main Content" and you should still see the dashboard.

> Type in some wrong paths off of the dashboard base path &page=stories-dashboard like &page=stories-dashboard#/jalkfjalkjflakdjflk and see that the dashboard default view is still showing.

> Also test a user that has no stories yet. What happens there is on the first page load the user is sent to "Explore Templates", the user interacting with "Main Content" for keyboard user is no longer the initial load so it'll actually bring the user to Dashboard Stories instead.

**Comment by felipebochehin87** [comment]
@BrittanyIRL the functionality is ok, but there's a minor tweak needed:

When I hit Enter on "Skip to main content" I do see the Dashboard, but "Dashboard" link is not "purpled out" (as if it has been clicked). Video below. That's the only missing detail.

https://images.zenhubusercontent.com/235435637/b28ffb92-90b3-4a9a-a876-1b84e1f6e016/2022_04_07_16_43_14.mp4

**Comment by BrittanyIRL** [comment]
Thanks for the catch! Will you look again? It should be fixed.

**Comment by felipebochehin87** [comment]
@BrittanyIRL I just noticed that the same happens when we enter an invalid URL (https://stories-qa-wordpress-amp.pantheonsite.io/wp-admin/edit.php?post_type=web-story&page=stories-dashboard#/jalkfjalkjflakdjflk) - the dashboard is displayed, but "Dashboard" link is not "purpled out" (as if it has been clicked).

1. Open the Dashboard
2. Click on Explore Templates
3. Enter an invalid URL such as https://stories-qa-wordpress-amp.pantheonsite.io/wp-admin/edit.php?post_type=web-story&page=stories-dashboard#/jalkfjalkjflakdjflk - Dashboard view is displayed, but the link is not purple.

**Comment by BrittanyIRL** [comment]
alright :) 3rd time's the charm! 🤞

**Comment by felipebochehin87** [comment]
Tested against https://stories-qa-wordpress-amp.pantheonsite.io/wp-admin/, using PR #11164.

1. Opened the Dashboard
2. Opened Explore Templates
3. Hit Tab until "Skip to main content" was displayed
4. Hit Enter - Dashboard displayed, "Dashboard" link purpled as if it has been clicked
5. Opened Explore Templates
6. Entered https://stories-qa-wordpress-amp.pantheonsite.io/wp-admin/edit.php?post_type=web-story&page=stories-dashboard#/jalkfjalkjflakdjflk - Dashboard displayed, "Dashboard" link purpled as if it has been clicked
